src/esp/gfx/PbrDrawable.cpp

```diff
@@ -117,31 +123,265 @@ void PbrDrawable::setMaterialValuesInternal(
                  "proper Metallic/Roughness texture pointers - either a "
                  "texture is expected but not present or vice versa.", );
 
-  if (materialData_->hasAttribute("normalTexturePointer")) {
+  if (const Cr::Containers::Optional<Mn::GL::Texture2D*> normalTexturePtr =
+          materialData_->findAttribute<Mn::GL::Texture2D*>(
+              "normalTexturePointer")) {
     flags_ |= PbrShader::Flag::NormalTexture;
-    matCache.normalTexture =
-        materialData_->attribute<Mn::GL::Texture2D*>("normalTexturePointer");
+    matCache.normalTexture = *normalTexturePtr;
     if (meshAttributeFlags_ & gfx::Drawable::Flag::HasTangent) {
       flags_ |= PbrShader::Flag::PrecomputedTangent;
     }
-    if (tmpMaterialData.normalTextureScale() != 1.0f) {
-      flags_ |= PbrShader::Flag::NormalTextureScale;
-      CORRADE_ASSERT(tmpMaterialData.normalTextureScale() > 0.0f,
-                     "PbrDrawable::PbrDrawable(): the normal texture scale "
-                     "must be positive.", );
-    }
+    // normal texture scale
+    matCache.normalTextureScale = tmpMaterialData.normalTextureScale();
+    CORRADE_ASSERT(tmpMaterialData.normalTextureScale() > 0.0f,
```

Comment:
I realize you didn't add this assert, but anyway it would be nice to convert it to an ESP_CHECK. Doesn't feel like an assert.

Reply:
Won't negative scales simply invert the normal texture, i.e. a weird but valid use case? I don't think this need an assertion at all.

Reply:
how about abs(matCache.normalTextureScale) > 0?

Reply:
oh, it won't bug out when I try to query it in the shader?  cool

Reply:
Why would it? You simply get a zero value, so as long as you don't divide anything with it, you'll get a valid result.